### PR TITLE
Capybara tests

### PR DESCRIPTION
**Pull request by emmanuellmhango**
In this branch I implemented the following:

- [ ]  solved the N+1 problem when fetching all posts and their comments for a user.
- before
log1

- After
log2

### CODE REVIEW

File: spec/integration/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  it 'should have the users number of posts' do
+    expect(page).to have_content('Number of posts: 1')
+    expect(page).to have_content('Number of posts: 2')
+  end
+
+  it 'should redirect to that users show page on click' do
+    click_link(@user1.name)
+    expect(page).to have_current_path user_path(@user1.id)
+  end
```
Comment (franclobo) [inline spec/integration/user_index_spec.rb:38]: - [ ] Kindly add a test to see the profile picture for each user for completing all the project requirements to approve the project 👇🏽 

> I can see the profile picture for each user.

File: spec/integration/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  it 'should have a button for seeing all posts' do
+    expect(page).to have_link('See All Posts')
+  end
+
+  it 'should redirect to all user posts index page on click' do
+    click_link('See All Posts')
+    expect(page).to have_current_path user_posts_path(@user.id)
+  end
```
Comment (franclobo) [inline spec/integration/user_show_spec.rb:45]: - [ ] Please 🙏🏽 add one more test for completing all the project requirements to approve the project 👇🏽 

> When I click a user's post, it redirects me to that post's show page.

File: spec/integration/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    expect(page).to have_content('Comments: 2')
+  end
+
+  it 'should have the number of likes for a post' do
+    expect(page).to have_content('Likes: 0')
+  end
+
+  it 'should have the body of a post' do
+    expect(page).to have_content('This is my first post!')
+  end
```
Comment (franclobo) [inline spec/integration/posts_index_spec.rb:52]: - [ ] There are two more tests to be added for completing all the project requirements to approve the project 👇🏽 

> I can see a section for pagination if there are more posts than fit on the view.
> When I click on a post, it redirects me to that post's show page.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @emmanuellmhango and @osbentech, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- N + 1 solved ✔️ 
- Capybara is used ✔️ 
- All integration tests passed ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/posts_index_spec.rb (hunk 3 above)
Comment (emmanuellmhango) [reply spec/integration/posts_index_spec.rb:52]: Dear @franclobo 

I have made the changes, please review

**Review by mwanawabangona — APPROVED** [review]
Hi @emmanuellmhango  & @osbentech 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  

H
## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**